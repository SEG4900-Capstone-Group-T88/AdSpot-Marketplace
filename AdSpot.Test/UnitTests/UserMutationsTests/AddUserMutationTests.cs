namespace AdSpot.Test.UnitTests.UserMutationsTests;

[Collection("adspot-inmemory-db")]
public class AddUserMutationTests
{
    private const string AddUserMutation = """
        mutation AddUser($input: AddUserInput!) {
            addUser(input: $input) {
                user {
                    userId
                    email
                    firstName
                    lastName
                }

                errors {
                    ... on Error {
                        __typename
                        message
                    }
                }
            }
        }
        """;

    [Fact]
    [Trait("Category", "Unit")]
    public async Task AddUserSuccessful()
    {
        var result = await TestServices.ExecuteRequestAsync(b =>
            b.SetQuery(AddUserMutation)
                .SetVariableValue(
                    "input",
                    new Dictionary<string, object?>
                    {
                        { "email", "newuser" },
                        { "password", "newuser" },
                        { "firstName", "new" },
                        { "lastName", "user" },
                    }
                )
        );

        result.MatchSnapshot();
    }

    [Fact]
    [Trait("Category", "Unit")]
    public async Task AddUserEmailAlreadyExists()
    {
        var result = await TestServices.ExecuteRequestAsync(b =>
            b.SetQuery(AddUserMutation)
                .SetVariableValue(
                    "input",
                    new Dictionary<string, object?>
                    {
                        { "email", TestDatabase.TestUser.Email },
                        { "password", TestDatabase.TestUser.Password },
                        { "firstName", TestDatabase.TestUser.FirstName },
                        { "lastName", TestDatabase.TestUser.LastName },
                    }
                )
        );

        result.MatchSnapshot();
    }
}
