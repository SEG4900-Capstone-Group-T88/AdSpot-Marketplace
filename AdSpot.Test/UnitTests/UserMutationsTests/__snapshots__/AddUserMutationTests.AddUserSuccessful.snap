{
  "data": {
    "addUser": {
      "user": {
        "userId": 2,
        "email": "newadspotuser@adspot.com",
        "firstName": "new",
        "lastName": "user"
      },
      "errors": null
    }
  }
}